sdk/servicebus/Microsoft.Azure.ServiceBus/src/Message.cs

```diff
@@ -99,6 +99,12 @@ public string PartitionKey
 
             set
             {
+                if (this.sessionId != null && !this.sessionId.Equals(value))
+                {
+                    // SessionId is set. Then partition key must be same as session id.
+                    throw new InvalidOperationException("PartitionKey:" + value + " is not same as SessionId:" + this.sessionId);
```

Comment:
nit: use string interpolation - $"PartitionKey: {value} is not same as SessionId: {this.sessionId}";